scripts/machine-launch-script.sh

```diff
@@ -1,83 +1,260 @@
+	# -b for non-interactive install
+	$SUDO bash ./install_conda.sh -b -p "$CONDA_INSTALL_PREFIX"
+    fi
+
+    # see https://conda-forge.org/docs/user/tipsandtricks.html#multiple-channels
+    # for more information on strict channel_priority
+    $SUDO "$CONDA_EXE" config --system --set channel_priority strict
+    # By default, don't mess with people's PS1, I personally find it annoying
+    $SUDO "$CONDA_EXE" config --system --set changeps1 false
+
+    # https://conda-forge.org/feedstock-outputs/ 
+    #	filterable list of all conda-forge packages
+    # https://conda-forge.org/#contribute 
+    #	instructions on adding a recipe
+    # https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/pkg-specs.html#package-match-specifications
+    #   documentation on package_spec syntax
+    CONDA_PACKAGE_SPECS=()
+
+
+    # https://docs.conda.io/projects/conda-build/en/latest/resources/compiler-tools.html#using-the-compiler-packages
+    #    for notes on using the conda compilers
+    CONDA_PACKAGE_SPECS=( gcc_linux-64 gxx_linux-64 )
+
+    # if we want to install prebuilt riscv toolchain, the bdutrosv channel has riscv-gnu-toolchain
+
+    # poky deps
+    CONDA_PACKAGE_SPECS+=( python=3.8 patch texinfo subversion chrpath git wget )
+    # qemu deps
+    CONDA_PACKAGE_SPECS+=( gtk3 glib pkg-config bison flex )
+    # qemu also requires being built against kernel-headers >= 2.6.38 because of it's use of
+    # MADV_NOHUGEPAGE in a call to madvise.  See:
+    # https://man7.org/linux/man-pages/man2/madvise.2.html
+    # https://conda-forge.org/docs/maintainer/knowledge_base.html#using-centos-7
+    # obvi this would need to be made linux-specific if we supported other MacOS or Windows
+    CONDA_PACKAGE_SPECS+=( "kernel-headers_linux-64>=2.6.38" )
+    # firemarshal deps
+    CONDA_PACKAGE_SPECS+=( rsync )
+    # cross-compile glibc 2.28+ deps
+    CONDA_PACKAGE_SPECS+=( make )
+    # other misc deps
+    # conda-forge splits out the shared library into libzlib
+    # not sure if we need both, but I'm including them for now
+    CONDA_PACKAGE_SPECS+=(
+        sbt \
+        ca-certificates \
+        mosh \
+        gmp \
+        mpfr \
+        mpc \
+        zlib \
+	libzlib \
+        vim \
+        git  \
+        openjdk \
+        gengetopt \
+        libffi \
+        expat \
+        libusb1 \
+        ncurses \
+        cmake \
+        graphviz \
+        expect
+        dtc \
+    )
+
+
+    # Why do we need these?
+    #   "perl(ExtUtils::MakeMaker)" \
+    # diffstat texi2html 
+
+    # verilator was added to conda-forge by timsnyder
+    CONDA_PACKAGE_SPECS+=( verilator==4.034 )
+
+    
+    # python packages
+    # While it is possible to install using pip after creating the
+    # conda environment, pip's dependency resolution can conflict with
+    # conda and create broken environments.  It's best to use the conda
+    # packages so that the environment is consistent
+    CONDA_PACKAGE_SPECS+=( \
+	boto3==1.20.21 \
```

Comment:
I'm not sure if these versions will change on a bump to the python version. Probably worthwhile to double-check.

Reply:
Sure.  If we bump python and the conda install fails because all of the requirements are unsolvable, it will tell us which ones have to be updated.